### DISCUSSION TITLE
withIAPContext makes App renders twice.

### DISCUSSION THREAD

**Discussion by dmk3141618** [hyochan/react-native-iap#2869] [Q&A]
Can I use withIAPContext on any component? not just App like

export default **withIAPContext**(App);

Can I use like this below if I only use purchase hooks on PurchaseItemScreen?

export default **withIAPContext**(PurchaseItemScreen);

**Description**

https://react-native-iap.hyo.dev/docs/api-reference/hooks

```typescript
import React from 'react';
import {withIAPContext} from 'react-native-iap';

const App = () => <View />;

export default withIAPContext(App);
```

**Expected Behavior**

Just render once. Because I have to watch app logs.

**Screenshots**

![image](https://github.com/user-attachments/assets/c0efc11a-b6ce-4dbf-a7bd-8c00528a35f5)

**Environment:**

- react-native-iap: 12.15.4
- react-native: 0.75.4
- Platforms (iOS, Android, emulator, simulator, device): iOS, Android

**To Reproduce**

Just run this code. And press 'r' to refresh App.

```typescript
import React from 'react';
import {withIAPContext} from 'react-native-iap';

const App = () => <View />;

export default withIAPContext(App);
```

---

[Optional] **Additional Context**

Is it because of this? (withIAPContext.tsx)

```typescript
    useEffect(() => {
      initConnection()
        .then((value) => {
          setInitConnectionError(undefined);
          setConnected(value);
        })
        .catch(setInitConnectionError);
    }, []);

    useEffect(() => {
      if (!connected) {
        return;
      }
```

**Comment by GiuseppeSpathis** [comment]
Try to remove react strict mode
import React from 'react';
import ReactDOM from 'react-dom';

ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  document.getElementById('root')
);

Remove <React.StrictMode>

**Reply by dmk3141618** [reply]
Thank you. But I am already not using StrictMode.
